frontend/src/libs/components/article-form/article-form.tsx

```diff
@@ -0,0 +1,103 @@
+import { ButtonType } from '~/libs/enums/button-type.enum.js';
+import { SubmitType } from '~/libs/enums/enums.js';
+import { useAppDispatch, useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+import { type ValueOf } from '~/libs/types/types.js';
+import {
+  articleCreateValidationSchema,
+  type ArticleRequestDto,
+} from '~/packages/articles/articles.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
+import { Button, Input, Textarea } from '../components.js';
+import { DEFAULT_ARTICLE_FORM_PAYLOAD } from './libs/constants.js';
+import styles from './styles.module.scss';
+
+const ArticleForm: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { control, errors, handleSubmit, reset, dirtyFields } =
+    useAppForm<ArticleRequestDto>({
+      defaultValues: DEFAULT_ARTICLE_FORM_PAYLOAD,
+      validationSchema: articleCreateValidationSchema,
+    });
+
+  const areFieldsDirty = !!dirtyFields.title && !!dirtyFields.text;
+
+  type SubmitType = ValueOf<typeof SubmitType>;
```

Comment:
1 - Don't place type declarations inside components
2 - Use type in `ValueOf<typeof SubmitType>` form, because now it looks confusing that here we can find SubmitType as enum and SubmitType as type
```suggestion

```

Reply:
Will it be correct, if I rename the type to ArticleSubmitTypeValue and move it to the types folder for article-form?

Reply:
Just use enum with `ValueOf<typeof [enum]>`